src/Hosting/Hosting/test/HostingApplicationDiagnosticsTests.cs

```diff
@@ -156,17 +156,61 @@ public void EventCountersEnabled()
     }
 
     [Fact]
-    public void MetricsEnabled()
+    public void Metrics_RequestDuration_RecordedWithHttpActivity()
     {
         // Arrange
-        var hostingEventSource = new HostingEventSource(Guid.NewGuid().ToString());
+        using var activityListener = new ActivityListener
+        {
+            ShouldListenTo = activitySource => activitySource.Name == "Microsoft.AspNetCore",
```

Comment:
You should create an activity source, inject it, and use reference equal so the test isn't affected by other tests.

See prior examples:
https://github.com/dotnet/aspnetcore/blob/bb25c6752dad2b99b06fd428feeeafb0bf4a2316/src/Hosting/Hosting/test/HostingApplicationDiagnosticsTests.cs#L835

Reply:
I don't think it matters for this test because it doesn't actually listen for activities. We're only listening so an activity is created.

But, for consistency and to reduce confusion, I'll update the test.